### DISCUSSION TITLE
Creating network map incredibly slow

### DISCUSSION THREAD

**Discussion by habitoti** [Koenkk/zigbee2mqtt#9775] [General]
I am currently migrating device by device from one Conbee 2 running deconz/phoscon to a second one running Zigbee2Mqtt. I just added 10 devices so far:

![2021-11-19_16-54-37](https://user-images.githubusercontent.com/220421/142754870-b7488375-6a0d-47b8-bdf5-258a77789e25.png)

For the first 4 or so devices, the network map was brought up quickly, but now with 10 devices it takes 5 (five!) minutes already. Once all 25 devices would be migrated, I'll suppose it will be unusable at all (it is effectively already now that way)

Is that to be expected, or is potentially something wrong with my setup? It's run on a PI 4 with SSD, which is generally pretty snappy running Homeassistant. Using latest Zigbee2Mqtt 1.22.0-3.

It also seems to be like all routes and link qualities are requested real time at the time of generating the map. Wouldn't it make sense to track and keep information over time when it becomes available (e.g. when passive sensor report) and then use that latest available data for the map when looking at it? I think deconz is doing it this way...at least the map (which is generally bad otherwise from a network viewer perspective...) always shows latest data instantly once you open the map.

**Comment by francisp2** [comment]
On my pi4 it takes 1m 16 seconds to show the map with 31 devices. But I'm not running the add-on.

**Reply by 1igeza** [reply]
You can speed up creating network map by comment 224 line in file lib/extension/networkMap.ts 
`//            await utils.sleep(1);  sleep 1 second between each scan to reduce stress on network.
`
For me it doesn't impact on stability of zegbee2mqtt
With commented this line and over 30 devices (9 routers) creating network map is taking about 13s for me.
```
Nov 22 07:45:45 Z2MQTT npm[11947]: Zigbee2MQTT:info  2021-11-22 07:45:45: Starting network scan (includeRoutes 'false')
...
Nov 22 07:45:58 Z2MQTT npm[11947]: Zigbee2MQTT:info  2021-11-22 07:45:58: Network scan finished

```

**Reply by habitoti** [reply]
Would I need to connect to the Docker to edit it (running Homeassistant on a PI...), or did you build the complete package?

**Reply by 1igeza** [reply]
I'm running it on ubuntu so I have ease access to files.

**Reply by habitoti** [reply]
So I commented that line, but looking at the code it would save 1sec per device. But as I have just 10 anyways, I am also not noticing any remarkable gain now. It's still several minutes, which is just ridiculous with so few devices so far. There must be some other catch lurking, however the log isn't indicating anything...

**Comment by SamMousa** [comment]
Could it be slow because one or more routers are not responding? And it just waits a long time before timing out?

**Comment by habitoti** [comment]
Probably yes, but I hope it's not busy waiting on responses at all. And my few routers are in fact all there and responding, so it shouldn't take 15 minutes for a network of ~20 devices. I hope...
